apps/core/src/hooks/useTotalFiatBalance.ts

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import { useCurrentAccount } from '@iota/dapp-kit';
+import { useBalance } from './useBalance';
+import { useTokenPrice } from './useTokenPrice';
+import { CoinFormat, formatBalance, useCoinMetadata } from './useFormatCoin';
+import { IOTA_TYPE_ARG } from '@iota/iota-sdk/utils';
+
+export function useTotalFiatBalance() {
+    const { data: { price } = {} } = useTokenPrice('iota');
+    const address = useCurrentAccount()?.address;
+    const { data: coinBalance } = useBalance(address!);
+    const totalBalance = Number(coinBalance?.totalBalance);
```

Comment:
To avoid future NaN  in the balance :trollface: 
```suggestion
    const totalBalance = Number(coinBalance?.totalBalance || 0);
```